Unify 4C Python project

davidrudlstorfer opened the pull request:
<!--
* Title: Provide a general summary of your changes in the Title above.

* Assignees:  If you know anyone who should likely handle bringing this pull request (PR) to completion -- that's usually yourself -- select them from the Assignees drop-down on the right.

* Labels: Update the label of the issue(s) addressed by this pull request to "Under Review".
-->

## Description and Context
<!--
Provide a brief and concise description of your proposed change. Questions you should think about:
* Why is this change required?  What problem does it solve?
* Is there a bigger picture? Is this PR a part of a larger set of changes? Which further steps are planned after merging this PR, if any?
* How has the proposed implementation been verified and tested?

Keep the description of the PR always up-to-date and concise.
-->

Before being able to simplify and unify the Python files in 4C it is beneficial to collect them in one place. Therefore, this PR proposes to move all Python files into our `four_c_python` "sub project". Additionally, a new pre-commit hook is introduced which checks that all Python files are located within this "sub project".

I suppose that a few checks will still fail 😆

## Related Issues and Pull Requests
<!--
If applicable, let us know how this pull request is related to any other open issues or pull requests by linking to them here.
Some suggestion for keywords:
Closes (will automatically close mentioned issue if merged), Blocks, Related to
-->

Follows #1287

gilrrei commented on utilities/four_c_python/src/four_c_development/clang_tidy_filter_output.py:
I am not sure if this should be in here. How is this executed?

sebproell replied:
I also stumbled over this. I think it is correct insofar as it is used for development purposes and static analysis. `clang-tidy` is in a weird place since it operates on the source folder but still requires a configured build to understand the compile commands.

gilrrei commented on doc/documentation/conf.py.in:
```diff
@@ -6,16 +6,11 @@
 import os
-# sys.path.append(os.path.abspath('.../site-packages/sphinx/ext'))
 # Include our own python helper modules
-sys.path.append('@PROJECT_SOURCE_DIR@/doc/documentation/python')
```
I think you have to add the path to the python packages below:

```python
sys.path.append('@PROJECT_SOURCE_DIR@/utilities/four_c_python/src')
```

ischeider reviewed (commented):
Since you added `four_c_documentation`, you have to add the respective information also to `four_c_python/README.md`. And I wonder whether we really would like to have the paraview scripts as executable tools. For me they were rather informations for users (and thus would belong to the `doc/documentation` directory. Of course, it would be great if we can make them work within our testing framework and maybe generate the figures automatically, but I don't see it currently.

davidrudlstorfer commented:
@gilrrei first of all thanks for the hints on how to fix the problems in the pipeline during the python setup

@gilrrei @ischeider sorry that this took so long, I needed to find some time to play around with these annoying errors in the pipeline 😅

I've rebased onto latest main and addressed all comments from above. As far as I can tell the pipeline should now hopefully run through and everything should work as expected.

Therefore, this would be ready for review 🚀
